src/cli/mod.rs

```diff
@@ -15,6 +15,23 @@ pub struct ParsecToolApp {
     #[structopt(short = "p", long = "provider")]
     pub provider: Option<u8>,
 
+    /// Sets the socket URI for communicating with the service.
+    ///
+    /// The value should be of the form "scheme://path". Currently the only supported
+    /// scheme is `unix`, signifying a Unix Domain Socket found at the given `path`
+    /// in the file system (e.g. `/some/path/parsec.sock`)
+    ///
+    /// If no value is provided, the following options are tried, in order:
+    ///
+    /// a) a socket URI stored in the `PARSEC_SERVICE_ENDPOINT` environment variable
+    ///
+    /// b) `unix:///run/parsec/parsec.sock` - the default socket location for the Parsec
+    /// Unix Domain Socket
+    ///
+    /// c) `unix:///tmp/parsec.sock` - a usual location used for test deployments
+    #[structopt(long, long = "socket_uri")]
+    pub service_socket_uri: Option<String>,
```

Comment:
I would be in favour to *only* have the environmental variable as a way to change the socket path. I personally think that the duplication of the environmental variable + the field is not needed.

Also, trying other usual paths is a nice feature for the user but I am worried about the security: setting `/tmp/parsec.sock` as a path is not secure in all cases and we would make that choice. I feel like if the user is the only one responsible for changing the path then they are the only one responsible for the vulnerabilities caused?

Reply:
> I would be in favour to only have the environmental variable as a way to change the socket path.

I was expecting you to say that - tbh I'm not sure which one to keep. Yes, using an environment variable is easier because you can just set it once and then use the tool without having to worry, but an argument seems more idiomatic for a CLI app. We don't take any other parameters through an environment variable, not sure why we'd make an exception for this one. I think the initial issue mentions them because it was written for the client, not for the tool.

> setting /tmp/parsec.sock as a path is not secure in all cases and we would make that choice.

True - alternatively we can print a big warning that we're using a socket in `/tmp` and they must be sure it's the correct one